src/main/java/de/tum/in/www1/artemis/service/ProgrammingExerciseTestCaseService.java

```diff
@@ -77,6 +78,13 @@ public ProgrammingExerciseTestCaseService(ProgrammingExerciseTestCaseRepository
             matchingTestCase.setBonusPoints(programmingExerciseTestCaseDTO.getBonusPoints());
             updatedTests.add(matchingTestCase);
         }
+
+        // Make sure that at least one test has a weight so that students can still achieve 100% score
+        var testWeightsSum = existingTestCases.stream().mapToDouble(ProgrammingExerciseTestCase::getWeight).sum();
```

Comment:
Will getWeight ever return null? I think this method will fail in this case. I am asking because I think in the client you are testinf if weight is undefined

Reply:
Done in `2b46a47`